### PR TITLE
spacedrive: 0.2.14 -> 0.3.1

### LINKED ISSUE #320338: Update request: spacedrive 0.2.14 → 0.3.1
- Package name: spacedrive
- Latest released version: [0.3.1](https://github.com/spacedriveapp/spacedrive/releases/tag/0.3.1)

- Current version on the unstable channel: 0.2.14
- Current version on the stable/release channel: 0.2.14

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@DataHearth @heisfer @mikaelfangel @stepbrobd

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by heisfer**
## Description of changes
resolves #320338 

release log: https://github.com/spacedriveapp/spacedrive/releases/tag/0.3.1
changelog: https://github.com/spacedriveapp/spacedrive/compare/0.2.14...0.3.1

Linux Note: Switched AppImage to deb.
Formatter: nixfmt-rfc-style

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MikaelFangel** [comment]
Result of `nixpkgs-review pr 320423` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 1]
```diff
@@ -1,73 +1,132 @@
...
+      src
+      meta
+      passthru
+      ;
+
+    sourceRoot = "Spacedrive.app";
+
+    nativeBuildInputs = [ undmg ];
+
+    installPhase = ''
```
Comment (MikaelFangel) [inline pkgs/by-name/sp/spacedrive/package.nix:76]: You're missing the runHook's here.

Comment (heisfer) [reply pkgs/by-name/sp/spacedrive/package.nix:76]: Oh, that was done by darwin maintainer. (it shows changed because of nixfmt-rfc-style format)

Comment (DataHearth) [reply pkgs/by-name/sp/spacedrive/package.nix:76]: Yup, mb, I totally forget it when I did it and it passed the review. But no longer cannot validate darwin packages. I no longer have a MacOS system at hands ^^.

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 320423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

**Review by DataHearth — APPROVED** [review]

File: pkgs/by-name/sp/spacedrive/package.nix (hunk 1 above)
Comment (stepbrobd) [reply pkgs/by-name/sp/spacedrive/package.nix:76]: That's probably me, sorry

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 320423` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

**Review by stepbrobd — APPROVED** [review]

**Review by DataHearth — APPROVED** [review]

**Pull request merged**